### PR TITLE
Publish error handling

### PR DESCRIPTION
**Pull request by Blackbaud-BobbyEarl**

### CODE REVIEW

File: test/lib-publish.spec.js [hunk 1]
```diff
@@ -60,6 +63,35 @@ describe('skyux-deploy lib publish', () => {
...
+    expect(loggerMock.info).toHaveBeenCalledWith(loggerInfoSuccess);
+  });
+
+  it('should log an error when settings.isStaticClient = false', async () => {
+    const error = 'some-error-happened-is-static-client-false';
+    publishSpaMock.and.callFake(() => {
+      throw error;
+    });
+
+    try {
```
Comment (Blackbaud-PaulCrowder) [inline test/lib-publish.spec.js]: I think you want to call `await expectAsync(...).toBeRejectedWith(...)` here.  The problem with the test as it's written is that if `callPublish()` doesn't throw an error, then no `expect`s get called and the test will pass.  See `lib-deploy.spec.js` for examples of how to use it. https://github.com/blackbaud/skyux-deploy/blob/master/test/lib-deploy.spec.js#L73

File: test/lib-publish.spec.js [hunk 2]
```diff
@@ -4,14 +4,28 @@ describe('skyux-deploy lib publish', () => {
...
+    return publish({
+      azureStorageAccessKey: 'abc',
+      isStaticClient: isStaticClient,
+      name: 'custom-name',
+      version: 'custom-version'
+    });
+  }
+
   beforeEach(() => {
+    loggerMock = jasmine.createSpyObj('logger', ['info', 'error', 'bobby']);
```
Comment (Blackbaud-PaulCrowder) [inline test/lib-publish.spec.js]: What's `bobby` used for?

File: test/lib-publish.spec.js (hunk 1 above)
Comment (Blackbaud-BobbyEarl) [reply test/lib-publish.spec.js]: Yup, I didn't a bad job of searching for `toBeRejectedWith`.  I'll update.

File: test/lib-publish.spec.js (hunk 2 above)
Comment (Blackbaud-BobbyEarl) [reply test/lib-publish.spec.js]: To maintain my presence in this repo.  J/K, I was debugging.  I'll remove it.

**Pull request merged**